src/controllers/company.number.controller.ts

```diff
@@ -0,0 +1,42 @@
+import { Session } from "ch-node-session-handler";
+import { NextFunction, Request, Response } from "express";
+import { ObjectionCompanyProfile } from "../model/objection.company.profile";
+import { OBJECTIONS_CONFIRM_COMPANY } from "../model/page.urls";
+import { getCompanyProfile } from "../services/company.profile.service";
+import { addToObjectionsSession, createObjectionsSession } from "../services/objections.session.service";
+import logger from "../utils/logger";
+
+/**
+ * POST controller for company number screen
+ * @param req
+ * @param res
+ * @param next
+ */
+const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    logger.debug("Attempt to retrieve and show the company details");
+
+    const companyNumber: string = req.body.companyNumber;
+    const session: Session = req.session as Session;
+    logger.info(`Retrieving company profile for company number ${companyNumber}`);
+    const token: string = getValidToken(session, req) as string;
+
+    const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+    if (session) {
+        createObjectionsSession(session);
+        addToObjectionsSession(session, "company_data", company);
+    }
+    return res.redirect(OBJECTIONS_CONFIRM_COMPANY);
+};
+
+const getValidToken = (session: Session, req: Request): string | undefined => {
```

Comment:
I would move this method into the new objections.session.service. I can see it being needed by other controllers in future and feels like it is a better 'fit' for this method.

Reply:
Moved and added unit tests.